src/ui/colortableview.js

```diff
@@ -152,45 +215,69 @@ export default class ColorTableView extends View {
 	 * Creates a static color table grid based on the editor configuration.
 	 *
 	 * @private
+	 * @returns {module:ui/src/colorgrid/colorgridview~ColorGridView}
 	 */
-	createStaticColorTable() {
+	_createStaticColorsGrid() {
 		const colorGrid = new ColorGridView( this.locale, {
 			colorDefinitions: this.colorDefinitions,
 			columns: this.columns
 		} );
 
 		colorGrid.delegate( 'execute' ).to( this );
-		colorGrid.bind( 'selectedColor' ).to( this );
 
 		return colorGrid;
 	}
 
 	/**
-	 * @inheritDoc
+	 * Creates document colors section view and binds it to {@link #documentColors}.
+	 *
+	 * @private
+	 * @returns {module:ui/src/colorgrid/colorgridview~ColorGridView}
 	 */
-	render() {
-		super.render();
+	_createDocumentColorsGrid() {
+		const documentColors = new ColorGridView( this.locale, {
+			columns: this.columns,
+			label: this.documentColorsLabel
+		} );
 
-		// Items added before rendering should be known to the #focusTracker.
-		for ( const item of this.items ) {
-			this.focusTracker.add( item.element );
-		}
+		documentColors.delegate( 'execute' ).to( this );
 
-		// Start listening for the keystrokes coming from #element.
-		this.keystrokes.listenTo( this.element );
-	}
+		documentColors.items.bindTo( this.documentColors ).using(
+			colorObj => {
+				const colorTile = new ColorTileView();
 
-	/**
-	 * Focuses the first focusable element in {@link #items}.
-	 */
-	focus() {
-		this._focusCycler.focusFirst();
-	}
+				colorTile.set( {
+					color: colorObj.color,
+					hasBorder: colorObj.options && colorObj.options.hasBorder
+				} );
 
-	/**
-	 * Focuses the last focusable element in {@link #items}.
-	 */
-	focusLast() {
-		this._focusCycler.focusLast();
+				if ( colorObj.label ) {
+					colorTile.set( {
+						label: colorObj.label,
+						tooltip: true
+					} );
+				}
+
+				colorTile.on( 'execute', () => {
+					this.fire( 'execute', {
+						value: colorObj.color,
+						label: colorObj.label,
+						options: Object.assign( {}, colorObj.options )
```

Comment:
Note that in my proposed changes there's no longer a need for `colorDefinition.options` so it's likely this code bit should be changed.